src/administration/horizontal_scaling.md

```diff
@@ -0,0 +1,141 @@
+- Outgoing federation
+- Scheduled tasks (most notably, constant hot rank calculations, which keep the front page fresh)
+
+By default, a Lemmy_server process will always run background scheduled tasks, which are intended to be run only on one
+server. Launching multiple processes with the default configuration will result in multiple duplicated scheduled tasks
+all starting at the same moment and trying to do the same thing at once. At best, it will be a waste of resources, but
+it could potentially end up causing some weird glitches as well.
+
+To solve this, Lemmy can be started with the `--disable-scheduled-tasks` flag on all but one instance. In general,
+there are two approaches:
+
+1. Run all your load balanced Lemmy servers with the `--disable-scheduled-tasks` flag, and run one additional Lemmy server
+   without this flag which is not in your load balancer and does not accept any HTTP traffic.
+2. Run one load balanced Lemmy server without the flag, and all other load balanced servers with the flag.
+
+Option 1 might have a few tiny advantages (easier to isolate logs for the scheduled tasks, and expensive scheduled tasks
+won't compete with HTTP requests for system resources), but requires a whole extra server.
+
+#### Rolling upgrades
+
+For most versions, rolling releases have been completely OK, but there have been some cases where a database migration
+slips in which is NOT backwards compatible (causing any servers running the older version of Lemmy to potentially start
+generating errors). To be safe, you should always first take a look at what database migrations are included in a new
+version, and evaluate whether the changes are for sure safe to apply with running servers.
+
+Note that even if there are no backwards incompatible migrations between immediate version upgrades (version 1 -> 2 is
+safe and 2 -> 3 is safe), doing bigger upgrades might cause these same migrations to become backwards incompatible
+(a server running version 1 might not be compatible with migrations from version 3 in this scenario). **It is generally
+best to only upgrade one version at a time!**.
+
+For the scheduled task process, it shouldn't cause any major issues to have two servers running in parallel for a short
+duration during an upgrade, but shutting the old one down before starting the new one will always be safer.
+
+### Pict-rs
+
+Pict-rs is a service which does image processing. It handles user-uploaded images as well as downloading thumbnails for
+external images.
+
+Pict-rs does not scale horizontally as of writing this document. This is due to a dependency on a Sled database, which
+is a disk based database file.
+
+The developer of pict-rs has mentioned plans to eventually add Postgres support (which should in theory enable
+horizontal scaling), but work on this has not yet started.
+
+Note that by caching pict-rs images in a CDN, you can really minimize load on the pict-rs server!
```

Comment:
Doesnt even need a CDN, you can do caching in nginx very effectively.

https://www.nginx.com/blog/nginx-caching-guide/

Reply:
I clarified this point